File: dir-client/src/http/user.http.ts
import { _URL } from "@/const"
import ky from "ky"

export const createCodeHttp = async(phone:string)=>{
    const create = await ky.post(`${_URL}/api/users/create-code`,{json:{phone}}).json()
    return create as string  
}

export const createUserHttp=async(phone:string,code:string)=>{
    const create = await ky.post(`${_URL}/api/users/create`,{json:{phone,code}}).json()
    return create as {id:number,hash:string}|null
}

export const getUserHttp=async(id:number)=>{
    const get = await ky.post(`${_URL}/api/users/get`,{json:{id}}).json()
    return get as {id:number,name:string,phone:string,dob:string}
}

export const updateUserHttp = async(body:{hash:string,id:number,name:string,dob:string})=>{
    const update = await ky.post(`${_URL}/api/users/update`,{json:body}).json()
    return update as {update:boolean}
}

File: flutter-client/http/user.http.dart
import 'dart:convert';

import 'package:delivery_app/tools/conts.dart';
import 'package:http/http.dart' as http;

import '../custom_classes/user.dart';

Future<String> createUserCodeHttp(String phone) async {
  var url = Uri.parse('$URL_MAIN/api/users/create-code');
  var req = await http.post(url, body:{'phone':phone});
  var decoded = jsonDecode(req.body);
  print(decoded);
  return (decoded as int).toString();
}

Future<Map<String, dynamic>> createUserHttp(String phone, String code) async {
  var url = Uri.parse('$URL_MAIN/api/users/create');
  print(code);
  var req = await http.post(url, body:{'phone':phone, 'code':code});
  var decoded = jsonDecode(req.body) as Map<dynamic, dynamic>;
  return decoded as Map<String, dynamic>;
}

Future<User> getUser(int id) async{
  var url = Uri.parse('$URL_MAIN/api/users/get');
  var req = await http.post(url, body:{'id': id.toString()});
  var decoded = jsonDecode(req.body) as Map<String, dynamic>;
  print(decoded);
  return User.fromJson(decoded);
}